crapssim/strategy/single_bet.py

```diff
@@ -238,6 +239,25 @@ def __repr__(self) -> str:
         )
 
 
+class BetHop(_BaseSingleBet):
+    def __init__(
+        self,
+        result: tuple[int, int],
+        bet_amount: typing.SupportsFloat,
+        mode=StrategyMode.ADD_IF_NOT_BET,
+    ):
+        if result[0] not in [1, 2, 3, 4, 5, 6] or result[1] not in [1, 2, 3, 4, 5, 6]:
+            raise NotImplementedError
+        self.result: tuple[int, int] = tuple(sorted(result))
+        super().__init__(Hop(result, bet_amount), mode=mode)
+
+    def __repr__(self) -> str:
+        return (
+            f"{self.__class__.__name__}({self.result}, bet_amount={self.bet.amount},"
+            f" mode={self.mode})"
+        )
```

Comment:
Implement `BetHop` strategy similar to other single bet strategies. Default mode is `ADD_IF_NOT_BET`.